velox/exec/trace/QueryMetadataReader.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/trace/QueryMetadataReader.h"
+
+#include "velox/common/file/File.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/connectors/hive/HiveDataSink.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/core/PlanNode.h"
+#include "velox/exec/PartitionFunction.h"
+#include "velox/exec/TableWriter.h"
+#include "velox/exec/trace/QueryTraceTraits.h"
+
+namespace facebook::velox::exec::trace {
+
+QueryMetadataReader::QueryMetadataReader(std::string traceOutputDir)
+    : pool_(memory::MemoryManager::getInstance()->tracePool()),
+      traceOutputDir_(std::move(traceOutputDir)) {
+  fileSystem_ = filesystems::getFileSystem(traceOutputDir_, nullptr);
+  VELOX_CHECK_NOT_NULL(fileSystem_);
+  if (!fileSystem_->exists(traceOutputDir_)) {
+    fileSystem_->mkdir(traceOutputDir_);
+  }
+  configFilePath_ = fmt::format(
+      "{}/{}", traceOutputDir_, QueryTraceTraits::kQueryConfigFileName);
+
+  Type::registerSerDe();
```

Comment:
This should be set in tool or test but not here.

Reply:
OK，fixed